### PR TITLE
Update sources to use AWS SDK Rust

### PR DESCRIPTION
**Pull request by rpkelly**
**Issue number:**
[1968](https://github.com/bottlerocket-os/bottlerocket/issues/1968)

**Description of changes:**
Updates pluto and cfsignal in sources to use AWS SDK Rust.

**Testing done:**
Unit testing and built image locally.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Comment by etungsten** [comment]
I've verified that the sundog/pluto proxy testing from https://github.com/bottlerocket-os/bottlerocket/pull/2227 still works as expected.

File: sources/cfsignal/src/cloudformation.rs [hunk 1]
```diff
@@ -1,10 +1,10 @@
+use std::str::FromStr;
+
```
Comment (etungsten) [inline sources/cfsignal/src/cloudformation.rs:2]: nit: extra line

File: sources/api/pluto/src/eks.rs [hunk 2]
```diff
@@ -1,33 +1,24 @@
...
-use std::str::FromStr;
 
 pub(crate) type ClusterNetworkConfig = KubernetesNetworkConfigResponse;
 
 #[derive(Debug, Snafu)]
 pub(super) enum Error {
     #[snafu(display("Error describing cluster: {}", source))]
     DescribeCluster {
-        source: RusotoError<DescribeClusterError>,
+        source: aws_sdk_eks::types::SdkError<aws_sdk_eks::error::DescribeClusterError>,
```
Comment (etungsten) [inline sources/api/pluto/src/eks.rs:16]: nit: I recommend some use declarations to make this path shorter. I think it'll help with readability a bit. This comment applies to other long import paths elsewhere as well.

**Review by etungsten — APPROVED** [review]

**Review by jpculp — APPROVED** [review]

File: sources/api/pluto/src/eks.rs (hunk 2 above)
Comment (etungsten) [reply sources/api/pluto/src/eks.rs:16]: on second thought, I think being explicit here is probably better

**Review by zmrow — APPROVED** [review]
:fondue:

File: sources/api/pluto/src/eks.rs [hunk 3]
```diff
@@ -57,6 +46,11 @@ pub(super) async fn get_cluster_network_config(
         .find(|env_var| *env_var != Err(env::VarError::NotPresent))
         .and_then(|s| s.ok());
 
+    let config = aws_config::from_env()
+        .region(Region::new(region.to_owned()))
```
Comment (etungsten) [inline sources/api/pluto/src/eks.rs:50]: I'm a little weirded out by the fact that `aws_sdk_rust` doesn't attempt to do validation on the input string. We also don't do any validation on `settings.aws.regions`. I'm assuming this just takes whatever is given here and generates API endpoints. Just something to keep in mind.

**Review by bcressey — APPROVED** [review]

File: sources/api/pluto/src/eks.rs (hunk 3 above)
Comment (rpkelly) [reply sources/api/pluto/src/eks.rs:50]: We could probably make some guesses surrounding what a region name looked like, but, I feel like we open ourselves to being broken every time a new region is released by doing so.

**Pull request merged**